src/ensembl/src/content/app/custom-download/components/content-builder/ContentBuilder.tsx

```diff
@@ -0,0 +1,272 @@
+
+type Path = (string | number)[];
+let path: Path = [];
+
+type ContentBuilderProps = {
+  data: AttributeWithContent;
+  selectedData: JSONValue;
+  onChange: (selectedData: JSONValue) => void;
+  contentState: JSONValue;
+  onContentStateChange: (updatedContentState: JSONValue) => void;
+  path?: Path;
+  contentProps?: {
+    [key: string]: JSONValue;
+  };
+};
+
+const ContentBuilder = (props: ContentBuilderProps) => {
+  if (!props.data) {
+    return null;
+  }
+
+  const onChangeHandler = (
+    path: (string | number)[],
+    payload: PrimitiveOrArrayValue
+  ) => {
+    const newSelectedData = { ...props.selectedData };
+    set(newSelectedData, path, payload);
+
+    props.onChange(newSelectedData);
+  };
+
+  const onContentStateChangeHandler = (
+    path: (string | number)[],
+    payload: PrimitiveOrArrayValue
+  ) => {
+    const updatedContentState = { ...props.contentState };
+    set(updatedContentState, path, payload);
+
+    props.onContentStateChange(updatedContentState);
+  };
+
+  const buildCheckboxWithSelect = (entry: AttributeWithOptions, path: Path) => {
+    const currentPath = [...path, entry.id];
+
+    const selectedOptions = get(props.selectedData, currentPath, []);
+
+    return (
+      <div className={styles.checkboxWithSelectWrapper}>
+        <CheckboxWithSelects
+          label={entry.label}
+          disabled={entry.disabled}
+          onChange={(selectedOptions: string[]) =>
+            onChangeHandler(currentPath, selectedOptions)
+          }
+          selectedOptions={selectedOptions}
+          options={(entry.options as Option[]) || []}
+        />
+      </div>
+    );
+  };
+
+  const buildCheckboxWithRadios = (entry: AttributeWithOptions, path: Path) => {
+    const currentPath = [...path, entry.id];
+
+    const selectedOption: string = get(props.selectedData, currentPath, '');
+
+    return (
+      <div className={styles.checkboxWithRadiosWrapper}>
+        <CheckboxWithRadios
+          label={entry.label}
+          disabled={entry.disabled}
+          onChange={(selectedOption: string | number | boolean) =>
+            onChangeHandler(currentPath, selectedOption)
+          }
+          selectedOption={selectedOption}
+          options={entry.options as RadioOptions}
+        />
+      </div>
+    );
+  };
+
+  const buildCheckboxGrid = (entry: AttributeWithOptions, path: Path) => {
+    const currentPath = [...path, entry.id];
+
+    const selectedOptions = get(props.selectedData, currentPath, []);
+
+    const newSelectedData = { ...props.selectedData };
+    const gridOptions = [...entry.options] as CheckboxGridOption[];
+    let shouldUpdateSelectedData = false;
+
+    gridOptions.forEach((option) => {
+      if (option.isChecked && selectedOptions[option.id] === undefined) {
+        set(newSelectedData, [...currentPath, option.id], true);
+        shouldUpdateSelectedData = true;
+      }
+    });
+    if (shouldUpdateSelectedData) props.onChange(newSelectedData);
```

Comment:
I think our style is to always use curly braces, even if it's just a one-liner.

Reply:
Cool 👍 updated it.